src/atomate2/abinit/flows/dfpt.py

```diff
@@ -0,0 +1,224 @@
+"""DFPT abinit flow makers."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING
+
+from jobflow import Flow, Maker
+
+from atomate2.abinit.jobs.core import StaticMaker
+from atomate2.abinit.jobs.mrgddb import MrgddbMaker
+from atomate2.abinit.jobs.response import (
+    DdeMaker,
+    DdkMaker,
+    DteMaker,
+    generate_dde_perts,
+    generate_ddk_perts,
+    generate_dte_perts,
+    run_dde_rf,
+    run_ddk_rf,
+    run_dte_rf,
+)
+from atomate2.abinit.powerups import (
+    update_factory_kwargs,
+    update_user_abinit_settings,
+    update_user_kpoints_settings,
+)
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+    from pymatgen.core.structure import Structure
+
+    from atomate2.abinit.jobs.base import BaseAbinitMaker
+
+
+@dataclass
+class DfptFlowMaker(Maker):
+    """
+    Maker to generate a DFPT flow with abinit.
+
+    The classmethods allow to tailor the flow for specific properties
+        accessible via DFPT.
+
+    Parameters
+    ----------
+    name : str
+        Name of the flows produced by this maker.
+    scf_maker : .BaseAbinitMaker
+        The maker to use for the static calculation.
+    ddk_maker : .BaseAbinitMaker
+        The maker to use for the DDK calculations.
+    dde_maker : .BaseAbinitMaker
+        The maker to use for the DDE calculations.
+    dte_maker : .BaseAbinitMaker
+        The maker to use for the DTE calculations.
+    mrgddb_maker : .Maker
+        The maker to merge the DDE and DTE DDB.
+    use_ddk_sym : bool
+        True if only the irreducible DDK perturbations should be considered,
+            False otherwise.
+    use_dde_sym : bool
+        True if only the irreducible DDE perturbations should be considered,
+            False otherwise.
+    dte_skip_permutations: Since the current version of abinit always performs
+        all the permutations of the perturbations, even if only one is asked,
+        if True avoids the creation of inputs that will produce duplicated outputs.
+    dte_phonon_pert: is True also the phonon perturbations will be considered.
+        Default False.
+    dte_ixc: Value of ixc variable. Used to overwrite the default value read
+        from pseudos.
+    """
+
+    name: str = "DFPT"
+    static_maker: BaseAbinitMaker = field(default_factory=StaticMaker)
```

Comment:
In the original dfpt workflow in abiflows, and in particular for the phonons the factory is `scf_for_phonons`, while here it would be the standard `scf_input`. Maybe we can keep using it here?
These could be achieved with either a new maker that sets uses that factory function (or sets stricter parameters as default) or with 
```python
static_maker: BaseAbinitMaker = field(default_factory=lambda: StaticMaker(factory=scf_for_phonons))
```

There can also be a `scf_for_dfpt` (that could be a copy or an alias for `scf_for_phonons`. I would keep the original one for compatibility with abiflows) and a specific ones for DTE if needed.

Reply:
I did not look into ```scf_for_phonons```, but it makes more sense to use it or smth similar indeed. I like the idea of having only one StaticMaker and just changing the factory. I will look into scf_for_phonons to see if it also suits the SHG case, if so then we can use it as is, otherwise we can switch to a modified copy of this factory.